plugin/src/_rules/border.js

```diff
@@ -0,0 +1,59 @@
+import { handler as h, directionMap, cornerMap } from "#utils";
+
+const borderStyles = [
+  "solid",
+  "dashed",
+  "dotted",
+  "double",
+  "hidden",
+  "none",
+  "groove",
+  "ridge",
+  "inset",
+  "outset",
+];
+
+export const borders = [
+  [/^border$/, handlerBorder],
+  [/^border()-(\d+)$/, handlerBorder, { autocomplete: "(border)-<directions>" }],
+  [/^border-([xy])-(\d+)$/, handlerBorder],
+  [/^border-([rltb])$/, handlerBorder],
+  [/^border-([rltb])-(\d+)$/, handlerBorder],
+  [/^border()-(.+)$/, handlerBorderStyle, { autocomplete: "(border)-style" }],
+  [/^border-([xy])-(.+)$/, handlerBorderStyle],
+  [/^border-([rltb])-(.+)$/, handlerBorderStyle],
+];
+
+function handlerBorder(m, ctx) {
+  const borderSizes = handlerBorderSize(m, ctx);
+  const borderStyle = handlerBorderStyle(["", m[1], "solid"]);
+  if (borderSizes && borderStyle) {
+    return [...borderSizes, ...borderStyle];
+  }
+}
+
+function handlerBorderSize([, a = "", b], { theme }) {
+  const v =
+    theme.lineWidth?.[b || "DEFAULT"] ?? h.bracket.cssvar.global.px(b || "1");
```

Comment:
We probably want to tighten this down. In "theme" we should define `lineWidth` to just those few values we want to support (in pixels).

Then we can nuke the `h` call down to `h.global(b)` - that way we support those global values of `unset` and blah blah.